### PR TITLE
Support national clouds

### PR DESCRIPTION
**Pull request by matthchr**
**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains tests

### CODE REVIEW

File: v2/internal/genericarmclient/generic_client.go [hunk 1]
```diff
@@ -11,9 +11,11 @@ import (
 	"strings"
 	"time"
 
-	"github.com/Azure/azure-service-operator/v2/internal/metrics"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
```
Comment (super-harsh) [inline v2/internal/genericarmclient/generic_client.go:14]: move this to line 20?

**Review by theunrepentantgeek — APPROVED** [review]

**Review by super-harsh — APPROVED** [review]

File: v2/internal/genericarmclient/generic_client.go (hunk 1 above)
Comment (matthchr) [reply v2/internal/genericarmclient/generic_client.go:14]: Done

**Pull request merged**